library/vcenter_gather_moids.py

```diff
@@ -78,6 +80,9 @@ def main():
             object_mo = find_cluster_by_name_datacenter(datacenter_mo, module.params[searched_parameter])
         elif searched_parameter == 'portgroup_name' and module.params[searched_parameter]:
             object_mo = get_mo(content, module.params[searched_parameter], VIM_TYPES[searched_parameter])
+        elif searched_parameter == 'virtualmachine_name' and module.params[searched_parameter]:
+            object_mo = get_mo(content, module.params[searched_parameter], VIM_TYPES[searched_parameter])
+            module.exit_json(changed=False, object_uuid=object_mo.config.instanceUuid)
```

Comment:
For attaching a VM to a switch or portgroup I need th UUID of the VM, nog the moId.

Normally you get an object from get_mo and returned the property _moId, but for the virtualmachine I would have the instanceUuid retrurned, which has no _moId. This is a workaround, but can be better.